Core/Controllers/UsersController.cs

```diff
@@ -214,5 +214,20 @@ public IActionResult Edit(string id, [FromForm, Bind] UserAccount accountChanges
                 ? RedirectToAction("me")
                 : RedirectToAction("display", new { id });
         }
+        
+        public IActionResult RefreshDataFromService(string id)
+        {
+            if (!User.Identity.IsAuthenticated)
+                return Unauthorized();
+
+            UserAccount targetUser = ProfileBuilder.FromId(id);
+            UserAccount selfUser = GetCurrentUser();
+
+            ProfileBuilder.UpdateDataFromService(User);
```

Comment:
Soo, update my data at all times, but if I'm not on my own page redirect me to the target user. That implies you can refresh your own details while on the profile page of another user?

Reply:
Oh right, I missed that that would be visible by admins.
Do you think that admins should be able to update the data for other users?

Reply:
No, that means we would have backwards information fetching on user's behalf. That goes against everything we did to prevent data scraping 😛

Reply:
Makes sense, I'll hide the button for non-self users then.

Reply:
Then I'd recommend to also hardcode it to prevent users from sending malicious packets.

Reply:
Hardcoding what exactly?

Reply:
if selfUser != targetUser return;

Reply:
Oh, yeah, good idea.